chainer/links/model/classifier.py

```diff
@@ -53,6 +54,8 @@ def __call__(self, x, t):
         self.accuracy = None
         self.y = self.predictor(x)
         self.loss = self.lossfun(self.y, t)
+        reporter.report({'loss': self.loss}, self)
```

Comment:
I am worried if it breaks backward compatibility because if we manually write training loop without the trainer object, which registers target links of optimizers registered to trainer, this link is not registered to reporter.